old_timesheet_migration/timesheet_migration.py

```diff
@@ -0,0 +1,212 @@
+
+sys.path.append('../')
+from config import settings
+from models import Activity, Client, Member, Project, TimeEntry
+
+def clean_timesheet(df, year):
+    '''Transform raw timesheet data frame to a data frame with a useful header, columns, and rows'''
+
+    header = df.iloc[3]
+    for i, head in enumerate(header):
+        if head == 'Soma':
+            last_index = i - 1
+            break
+    dates = header[5:last_index].apply(lambda x: x[:5].replace('/', '-') + '-' + year)
+    header = header[:5].append(dates)
+    df = df[4:][df.columns[:last_index]]
+    df = df.rename(columns=header)
+    df = df.drop(['Curso'], axis=1)
+    df = df[df['Integrante'].notnull()]
+    df.update(df.fillna(0))
+    return df
+
+def read_timesheet(year_semester, path):
+    '''Read the timesheet of year and semester in a Path
+        Returns a dataframe.'''
+
+    timesheet_file = pd.ExcelFile(path)
+    raw_timesheet = pd.read_excel(timesheet_file, 'Timesheet ' + year_semester[:4] + '.' + year_semester[4])
+    timesheet = clean_timesheet(raw_timesheet, year_semester[:4])
+    
+    return timesheet
+
+def row_valid(project_name, activity_name, client_name, no_time_projects):
+    '''Check if is a valid row based in project, activity and client.'''
+    
+    if project_name in no_time_projects:
+        return True
+
+    if 0 in (project_name, activity_name, client_name):
+        return False
+
+    return True
+
+def calculate_start_end(time, column):
+    '''Receive time and the day 
+        Returns the start and end in correct format.'''
+    
+    time_hours = (time * 50) // 60
+    time_minutes = (time * 50) % 60
+    start = datetime.strptime(column + 'T00:00:00-0300', "%d-%m-%YT%H:%M:%S%z")
+    start = start - timedelta(days=1)
+    end = start + timedelta(hours=time_hours, minutes=time_minutes)
+
+    return (start, end)
+
+def get_time(time):
+    '''Returns a float that represents the time.'''
+
+    if type(time) is str:
+        try:
+            time = float(time.replace('ha', '').replace(',', '.'))
+        except:
+            time = 0
+
+    return time
+
+def create_time_entries(time_entries, clean_timesheet):
+    '''Returns a dataframe with all time entries in path file.'''
+
+    timesheet_collumns = list(clean_timesheet.iloc[:, 4:])
+    for key, row in clean_timesheet.iterrows():
+        member_acronym = row[0].lower()
+        project_name = row[1]
+        activity_name = row[2]
+        client_name = row[3]
+
+        no_time_projects = ('Feriado', 'Falta_Justificada')
+        if not row_valid(project_name, activity_name, client_name, no_time_projects):
+            time_entries.drop(key, inplace=True)
+
+        else:
+            if project_name in no_time_projects:
+                activity_name = ''
+                client_name = 'no time'
+
+            else:
+                activity_name = activity_name.lower()
+                client_name = client_name.lower()
+
+            project_name = project_name.lower()
```

Comment:
I feel that all this could be a function. What exactly does this part of the code do? Does it fix acitivity_name, client_name and project_name?